src/EditorFeatures/Core/Implementation/FindReferences/DocumentLocation.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Navigation;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.FindReferences
+{
+    /// <summary>
+    /// Represents a <see cref="TextSpan"/> location in a <see cref="Document"/>.
+    /// </summary>
+    internal struct DocumentLocation : IEquatable<DocumentLocation>, IComparable<DocumentLocation>
+    {
+        public Document Document { get; }
+        public TextSpan SourceSpan { get; }
+
+        public DocumentLocation(Document document, TextSpan sourceSpan)
+        {
+            Document = document;
+            SourceSpan = sourceSpan;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return Equals((DocumentLocation)obj);
+        }
+
+        public bool Equals(DocumentLocation obj)
+        {
+            return this.CompareTo(obj) == 0;
+        }
+
+        public static bool operator ==(DocumentLocation d1, DocumentLocation d2)
+        {
+            return d1.Equals(d2);
+        }
+
+        public static bool operator !=(DocumentLocation d1, DocumentLocation d2)
+        {
+            return !(d1 == d2);
+        }
+
+        public override int GetHashCode()
+        {
+            return Hash.Combine(this.Document.FilePath, this.SourceSpan.GetHashCode());
```

Comment:
Your hashcode is case sensitive, but your `CompareTo` is not.

In general, I'm not sure if you actually want to be case-insenstive here though.  Consider the case where case is *changed*.  You'd want those to compare differently.  Since this is used for presentation, I generally prefer being case sensitive right up until the point where you access the file-system.